Gems/ROS2/Code/Source/RobotImporter/SDFormat/Hooks/ROS2SkidSteeringModelHook.cpp

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ROS2/Frame/ROS2FrameComponent.h>
+#include <RobotControl/Controllers/SkidSteeringController/SkidSteeringControlComponent.h>
+#include <RobotControl/ROS2RobotControlComponent.h>
+#include <RobotImporter/SDFormat/ROS2ModelPluginHooks.h>
+#include <RobotImporter/SDFormat/ROS2SDFormatHooksUtils.h>
+#include <Source/EditorArticulationLinkComponent.h>
+#include <Source/EditorHingeJointComponent.h>
+#include <VehicleDynamics/ModelComponents/SkidSteeringModelComponent.h>
+#include <VehicleDynamics/Utilities.h>
+
+#include <sdf/Joint.hh>
+#include <sdf/Link.hh>
+#include <sdf/Model.hh>
+#include <sdf/Plugin.hh>
+
+namespace ROS2::SDFormat
+{
+    namespace Parser
+    {
+        AZ::EntityId GetAxleWheelId(const std::string& jointName, const sdf::Model& sdfModel, const CreatedEntitiesMap& createdEntities)
+        {
+            const auto jointPtr = sdfModel.JointByName(jointName);
+            if (jointPtr != nullptr)
+            {
+                const auto linkName(jointPtr->ChildName().c_str());
+                const auto linkPtr = sdfModel.LinkByName(linkName);
+                if (linkPtr != nullptr && createdEntities.contains(linkPtr))
+                {
+                    const auto& entityResult = createdEntities.at(linkPtr);
+                    return entityResult.IsSuccess() ? entityResult.GetValue() : AZ::EntityId();
+                }
+            }
+
+            return AZ::EntityId();
+        }
+
+        void EnableMotor(const AZ::EntityId& entityId)
+        {
+            AZ::Entity* entity = nullptr;
+            AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+            if (entity != nullptr)
```

Comment:
Switching to early return here will improve readability.

Reply:
My idea was to keep _hooks_ independent from the rest of the code. I do not want to pass too many parameters to the creation callback either to keep the code simple and easy to implement in external Gems. This means there is no information in the _hook_  if the importer uses the articulations or not.

The current flow is following:
1) assume _no articulation_ and set _enable motor_ in joint; terminate if succeed
2) assume _articulation_ and add _WheelComponent_ required in _use articulation_ configuration; terminate if succeed
3) print warning if both failed

I can move the warning and early return before, but it would still require an early return after 1)